config/default.yml

```diff
@@ -20,27 +20,52 @@ AllCops:
   DisplayStyleGuide: true
   DisplayCopNames: true
 
+Betterment:
+  StyleGuideBaseURL: https://github.com/Betterment/betterlint
+
 Betterment/ServerErrorAssertion:
   Description: 'Detects assertions on 5XX HTTP statuses.'
   Include:
     - 'spec/requests/**/*_spec.rb'
 
 Betterment/AuthorizationInController:
   Description: 'Detects unsafe handling of id-like parameters in controllers.'
+  StyleGuide: '#bettermentauthorizationincontroller'
   Enabled: false
 
 Betterment/UnsafeJob:
   Enabled: false
+  StyleGuide: '#bettermentunsafejob'
   sensitive_params:
     - password
     - social_security_number
     - ssn
 
+Betterment/UnscopedFind:
+  StyleGuide: '#bettermentunscopedfind'
+
+Betterment/DynamicParams:
+  StyleGuide: '#bettermentdynamicparams'
+
 Betterment/SitePrismLoaded:
   Include:
     - 'spec/features/**/*_spec.rb'
     - 'spec/system/**/*_spec.rb'
 
+Betterment/NonStandardActions:
+  Description: 'Detects non-standard controller actions.'
+  StyleGuide: '#bettermentnonstandardactions'
+  AllowedActions:
+    - index
+    - show
+    - new
+    - edit
+    - create
+    - update
+    - destroy
```

Comment:
I was not aware of this! I've checked and there are 9 examples of this in cst. I don't see any in retail, institutional, or shrink.

I've been thinking about it, and I'd like to bring it to the Eng Forum, but personally I think we should stick to the default 7. My reasoning is that I think the important aspect of this change is standardizing as an org on what actions we use and I think the usefulness of the `delete` action isn't worth the ongoing cost of training engineers on a non-default list of actions.

One of the motivating examples for this cop was one surrounding impersonation, where by default we only allow 'index', 'show, 'new', and 'edit'. The `delete` action is not included on this list so if someone added that action, impersonators would not be able to view it. We could add `delete` to the list, but it just underscores to me how we need everyone to be on the same page on this list.

What do you think?

Reply:
I'm fine coming up with another pattern that replaces `delete` with a standard action! It seems like the pattern didn't really catch on outside of those use cases in cst, so I don't think we are losing a lot by not including it in this list for now. If it turns out we really do want to bring it back, we can always figure something out.

Reply:
sgtm. This cop does per-app configuration if we ever decided to do that. thanks

Reply:
@medlefsen what does the per-app config look like if we need to add `delete` as an allowed operation?

(btw, i'm a huge fan of `delete` as a pattern -- feels like Rails has always been missing the kind of "deletion preview" that other frameworks like ASP.NET MVC have -- but I agree that we'd need to align on it more broadly before it would make sense as a default)

Reply:
I ask because we've seen that rubocop YMLs don't compose neatly -- so I'm thinking that the consuming app will have to redefine the entire default config. It would be nice if there were a way to let clinic define just `delete` as an additional allowed action, but that might involve pulling more of the defaults into the Ruby implementation of the cop, if that makes sense. 🤔

Reply:
Ah yeah, that's a good thing to check. I just added this to cst's config:

```yaml
Betterment/NonStandardActions:
  AllowedActions:
    - index
    - show
    - new
    - edit
    - create
    - update
    - delete
    - destroy
```

and then ran rubocop, and it looks like it worked:

```
Offenses:

config/routes.rb:23:5: C: Betterment/NonStandardActions: Route goes to a non-standard controller action. Use a new controller instead of custom actions. (https://github.com/Betterment/betterlint#bettermentnonstandardactions)
    post :toggle_manual_bank_linking_enabled_at, controller: 'retail_users', action: 'toggle_manual_bank_linking_enabled_at'
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

604 files inspected, 1 offense detected
```

Reply:
Maybe just a minor thing, but I wonder if the UX could encourage folks to add to the list, rather than rewrite the entire list:

```yml
Betterment/NonStandardActions:
  AllowedActions:
    - delete
```

My thought is that it would be _very_ unlikely (and probably an anti-pattern) if anyone ever attempted to _remove_ on of Rails' standard actions from that list.